File: sections/03_counting_frequencies/problems/b_frequency_winner/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;
    while (T--) {
        int n;
        cin >> n;
        unordered_map<int, int> freq;
        freq.reserve(2 * n + 1);
        for (int i = 0; i < n; ++i) {
            int x;
            cin >> x;
            ++freq[x];
        }

        int best_value = 0;
        int best_count = -1;
        for (auto [value, count] : freq) {
            if (count > best_count || (count == best_count && value < best_value)) {
                best_value = value;
                best_count = count;
            }
        }
        cout << best_value << ' ' << best_count << '\n';
    }

    return 0;
}

File: sections/03_counting_frequencies/problems/b_frequency_winner/solution.py
"""Reference solution for B. Frequency Winner."""

from collections import Counter
import sys


def winner(values: list[int]) -> tuple[int, int]:
    freq = Counter(values)
    best_value = values[0]
    best_count = -1
    for value, count in freq.items():
        if count > best_count or (count == best_count and value < best_value):
            best_value = value
            best_count = count
    return best_value, best_count


def main() -> None:
    data = list(map(int, sys.stdin.buffer.read().split()))
    if not data:
        return
    t = data[0]
    idx = 1
    out: list[str] = []
    for _ in range(t):
        n = data[idx]
        idx += 1
        values = data[idx:idx + n]
        idx += n
        value, count = winner(values)
        out.append(f"{value} {count}")
    sys.stdout.write("\n".join(out))


if __name__ == "__main__":
    main()

